src/interpreter.cc

```diff
@@ -668,11 +668,15 @@ DEFINE_BITCAST(bitcast_u64_to_f64, uint64_t, double)
     }                                                           \
   } while (0)
 
-#define UNOP_FLOAT(type, func)                                 \
-  do {                                                         \
-    FLOAT_TYPE_##type value = BITCAST_TO_##type(POP_##type()); \
-    PUSH_##type(BITCAST_FROM_##type(func(value)));             \
-    break;                                                     \
+#define UNOP_FLOAT(type, func)                                   \
+  do {                                                           \
+    FLOAT_TYPE_##type value = BITCAST_TO_##type(POP_##type());   \
+    VALUE_TYPE_##type result = BITCAST_FROM_##type(func(value)); \
+    if (WABT_UNLIKELY(IS_NAN_##type(result))) {                  \
```

Comment:
There are a few other places where we explicitly set the quiet nan bit; since it's implementation defined what happens with NaNs AIUI, it's probably better to be explicit.